### PR TITLE
F64 support

### PR DESCRIPTION
**Pull request by c-git**
- Fix typo in i64
- Add f64 and Vec<f64> as known types
- Add approximately equal checking to 5 decimal places on problems where return type is f64

### CODE REVIEW

File: src/tool/core/helpers/problem_code.rs [hunk 1]
```diff
@@ -136,6 +136,15 @@ impl FunctionInfo {
         names.join(", ")
     }
 
+    pub fn get_solution_comparison_code(&self) -> String {
+        if let Some(FunctionArgType::F64) = &self.return_type {
+            "assert!((actual - expected).abs() < 1e-5, \"Assertion failed: actual {actual:.5} but expected {expected:.5}. Diff is more than 1e-5.\");"
+        } else {
```
Comment (sak96) [inline src/tool/core/helpers/problem_code.rs:142]:
```rust
else if  Some(FunctionArgType::VecF64) = &self.return_type {
            r#"
            assert!(actual.len(), expected.len(), "actual {actual} not same length as {expected}");
            actual.iter().zip(expected.iter()).enumerate().for_each(|(i, (actual, expected))| {
            assert!((actual - expected).abs() < 1e-5, "Assertion failed: {i} the element in actual {actual:.5} but expected {expected:.5}. Diff is more than 1e-5.");
           })
            "#
} else
```

Comment (sak96) [reply src/tool/core/helpers/problem_code.rs:142]: may be match statement is better ?

**Review by sak96 — CHANGES_REQUESTED** [review]

File: src/tool/core/helpers/problem_code.rs (hunk 1 above)
Comment (c-git) [reply src/tool/core/helpers/problem_code.rs:142]: That's what I had and then got recommended to use an if let by clippy. When we add more we'll change it to a match for now I think this is fine

Comment (c-git) [reply src/tool/core/helpers/problem_code.rs:142]: > ```rust
> else if  Some(FunctionArgType::VecF64) = &self.return_type {
>             r#"
>             assert!(actual.len(), expected.len(), "actual {actual} not same length as {expected}");
>             actual.iter().zip(expected.iter()).enumerate().for_each(|(i, (actual, expected))| {
>             assert!((actual - expected).abs() < 1e-5, "Assertion failed: {i}th element in actual {actual:.5} but expected {expected:.5}. Diff is more than 1e-5.");
>            })
>             "#
> } else
> ```

Oh you are doing both. Yeah better to change it to a match. Is there a question where that is the return type?

Comment (sak96) [reply src/tool/core/helpers/problem_code.rs:142]: created #33! resolving

**Review by sak96 — APPROVED** [review]

**Pull request merged**